### DISCUSSION TITLE
how do i SetVisibility of a mesh created with pv.MultiBlock()?

### DISCUSSION THREAD

**Discussion by JohnForeman** [pyvista/pyvista#4218] [Q&A] [answered]
I have this code that uses an array of plane definitions.  I'm using pv.MultiBlock because chatgpt suggested i use it to get past an error i was having simply trying to add the mesh with the array.  I'm loading another mesh from an stl file and the toggle visibility works fine for it.  Can someone see my mistake or suggest a different way of creating the planes so that that the toggle works?

```py
planes = []
planes_lab = []
planes_lab_loc = []
for i in range(plane_nom_ary.shape[0]):
    center = plane_nom_ary[i, 1:4].astype(float)
    direction = plane_nom_ary[i, 4:7].astype(float)
    plane = pv.Plane(center=center, direction=direction, i_size=30, j_size=30)
    planes.append(plane)
    planes_lab_loc.append(center)
    planes_lab.append(plane_nom_ary[i, 0])

merged_planes = pv.MultiBlock(planes)
plane_mesh = plotter.add_mesh(merged_planes,show_edges=True, opacity=1, line_width=5, color='#00FF00', style='wireframe')
plane_labels = plotter.add_point_labels(planes_lab_loc, planes_lab, show_points=False, point_color='blue', shape_opacity=0, font_family= 'arial',font_size=18, point_size=5, tolerance=.0001, text_color="blue")
```

when i try to toggle the visibility with **plane_mesh.SetVisibility(not plane_mesh.GetVisibility())** i get this error
**plane_mesh.SetVisibility(not plane_mesh.GetVisibility())
AttributeError: 'tuple' object has no attribute 'SetVisibility'**

**Comment by banesullivan** [comment, marked as answer]
When passing a `MultiBlock` object, unnder the hood, `add_mesh` using `add_composite`. `add_composite` returns both the actor and mapper (the tuple). 

Try:

```
plane_mesh, _ = plotter.add_mesh(merged_planes, ...)
```

@pyvista/developers, IMO this return type is a problem that needs to be resolved.

**Reply by banesullivan** [reply]
https://docs.pyvista.org/api/plotting/_autosummary/pyvista.Plotter.add_composite.html#add-composite

**Reply by adeak** [reply]
Indeed seems like a bug to me. Discard the mapper in `add_mesh()` and maybe add a note in its docs that `add_composite()` can be used if one needs the mapper.

**Reply by banesullivan** [reply]
FYI, #4261
